lib/angular/fieldLocation/fieldLocationController.js

```diff
@@ -15,19 +15,43 @@ function FieldLocationController($scope) {
   fieldLocationCtrl.setLocation = function(event) {
     event.stopPropagation();
 
-    //TODO Move it to a service to get the location
-    navigator.geolocation.getCurrentPosition(function(pos) {
+    /*
+    * Get the current position from the Geolocation Service.
+    */
+    GeolocationService.getGeolocation(function(err, response) {
+      if (response) {
+        console.log('Position Obtained');
+        fieldLocationCtrl.applyValues(response.coords.latitude, response.coords.longitude);
+      } else {
+        console.log(err.message ? err.message : err);
+        // location not available, set a default
+        fieldLocationCtrl.applyValues(-1, -1);
+      }
+    });
+
+    /**
+    * Apply the longitude/latitude on the scope.
+    * @param latitude - the latitude to set
+    * @param longitude - the longitude to set
+    */
+    fieldLocationCtrl.applyValues = function(latitude, longitude) {
       $scope.$apply(function() {
-        fieldLocationCtrl.field.value.lat = parseFloat(pos.coords.latitude);
-        fieldLocationCtrl.field.value.long = parseFloat(pos.coords.longitude);
-        console.log('position set', fieldLocationCtrl.field.value);
+        fieldLocationCtrl.field.value.lat = parseFloat(latitude);
+        fieldLocationCtrl.field.value.long = parseFloat(longitude);
       });
-    }, function() {
-      console.log("Location Not available.");
-      fieldLocationCtrl.field.value.lat = -1;
-      fieldLocationCtrl.field.value.long = -1;
-    });
+    };
+
+    /*
+    * Update values on the input change
+    */
+    fieldLocationCtrl.updateModel = function() {
+      console.log("Updated Model");
+      fieldLocationCtrl.field.value.lat = $scope.field.values[0].lat;
```

Comment:
These should be the other way around:

`fieldLocationCtrl.field.values[0] = fieldLocationCtrl.field.value`